src/sidebars/search/search-controller.js

```diff
@@ -78,69 +86,83 @@
             this.noResults = false;
 
             this.currentWindowService.ready(() => {
-                if (!this.store) {
-                    this.store = window.storeService.open(window.name);
-                }
-
-                var favourites = this.store.get();
-                if (this.query) {
-                    var length = favourites.length;
-                    this.isLoading = true;
-                    this.errors = [];
-                    this.quandlService.search(this.query, (stock) => {
-                        this.isLoading = false;
-                        var i;
-
-                        // removing stocks found with old query
-                        this.stocks = this.stocks.filter((result, j) => {
-                            return result.query === this.query;
-                        });
-
-                        // not adding old stocks
-                        if (stock.query !== this.query) {
-                            return;
-                        }
-
-                        // Due to the asynchronicity of the search, if multiple searches
-                        // are fired off in a small amount of time, with an intermediate one
-                        // returning no results it's possible to have both the noResults flag
-                        // set to true, while some stocks have been retrieved by a later search.
-                        //
-                        // Here we re-set the flag to keep it up-to-date.
-                        this.noResults = false;
-
-                        var stockAdded = false;
-                        for (i = 0; i < length; i++) {
-                            if (stock.code === favourites[i]) {
-                                stock.favourite = true;
-                                this.stocks.unshift(stock);
-                                stockAdded = true;
+                if (this.hasStore()) {
+                    var favourites = this.store.get();
+                    if (this.query) {
+                        var length = favourites.length;
+                        this.isLoading = true;
+                        this.errors = [];
+                        this.quandlService.search(this.query, (stocks) => {
+                            this.isLoading = false;
+
+                            // removing stocks found with old query
+                            this.stocks = this.stocks.filter((result) => {
+                                return result.query === this.query;
+                            });
+
+                            // Only show the favourites if the query is empty
+                            if (this.query.trim() === '') {
+                                this.displayFavourites();
```

Comment:
Probably not. I'll move the favourites var down to where it's used (as part of the above refactor with `.forEach`).